worldedit-bukkit/adapters/adapter-1_20_2/src/main/java/com/sk89q/worldedit/bukkit/adapter/ext/fawe/v1_20_R2/PaperweightWorldNativeAccess.java

```diff
@@ -38,20 +39,50 @@
 import java.lang.ref.WeakReference;
 import java.util.Objects;
 import javax.annotation.Nullable;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
 
 public class PaperweightWorldNativeAccess implements WorldNativeAccess<LevelChunk, net.minecraft.world.level.block.state.BlockState, BlockPos> {
     private static final int UPDATE = 1;
     private static final int NOTIFY = 2;
 
     private final PaperweightAdapter adapter;
     private final WeakReference<ServerLevel> world;
+    private final MethodHandle globalTickData;
+    private final Class<?> regionScheduleHandleClass;
+    private final Class<?> regionizedServerClass;
+    private final MethodHandle globalCurrentTick;
     private SideEffectSet sideEffectSet;
 
     public PaperweightWorldNativeAccess(PaperweightAdapter adapter, WeakReference<ServerLevel> world) {
         this.adapter = adapter;
         this.world = world;
+        PaperweightAdapter adapter = (PaperweightAdapter) paperweightFaweAdapter.getParent();
+        if (adapter.isFolia()) {
+            try {
+                regionizedServerClass = Class.forName("io.papermc.paper.threadedregions.RegionizedServer");
+                regionScheduleHandleClass = Class.forName(
+                        "io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle");
+                globalTickData = MethodHandles.lookup().unreflect(regionizedServerClass.getDeclaredMethod("getGlobalTickData"));
+                var data = globalTickData.invoke();
+                globalCurrentTick = MethodHandles.lookup().unreflect(regionScheduleHandleClass.getDeclaredMethod(
+                        "getCurrentTick"));
+                final long tick = (long) globalCurrentTick.invoke(data);
+                this.lastTick = new AtomicInteger((int) tick);
+            } catch (Throwable e) {
+                throw new RuntimeException(e);
+            }
+
+        } else {
```

Comment:
Transfer that code into default blocks into the interface class. That reduces the code